pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+from typing import (
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    FrameOrSeries,
+    PositionalIndexer,
+)
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+from pandas.core.groupby import groupby
+from pandas.core.indexes.api import CategoricalIndex
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .rows to GroupBy.
+    """
+
+    @cache_readonly
+    def _rows(self) -> _rowsGroupByIndexer:
+        return _rowsGroupByIndexer(cast(groupby.GroupBy, self))
+
+
+@doc(GroupByIndexingMixin._rows)
+class _rowsGroupByIndexer:
+    def __init__(self, groupByObject: groupby.GroupBy):
+        self.groupByObject = groupByObject
+
+    def __getitem__(self, arg: PositionalIndexer | tuple) -> FrameOrSeries:
+        """
+        Positional index for selection by integer location per group.
+
+        Used to implement GroupBy._rows which is used to implement GroupBy.nth
+        when keyword dropna is None or absent.
+        The behaviour extends GroupBy.nth and handles DataFrame.groupby()
+        keyword parameters such as as_index and dropna in a compatible way.
+
+        The additions to nth(arg) are:
+        - Handles iterables such as range.
+        - Handles slice(start, stop, step) with
+            start: positive, negative or None.
+            stop: positive, negative or None.
+            step: positive or None.
+
+        Parameters
+        ----------
+        arg : PositionalIndexer | tuple
+            Allowed values are:
+            - Integer
+            - Integer values iterable such as list or range
+            - Slice
+            - Comma separated list of integers and slices
+
+        Returns
+        -------
+        Series
+            The filtered subset of the original groupby Series.
+        DataFrame
+            The filtered subset of the original groupby DataFrame.
+
+        See Also
+        --------
+        DataFrame.iloc : Purely integer-location based indexing for selection by
+            position.
+        GroupBy.head : Return first n rows of each group.
+        GroupBy.tail : Return last n rows of each group.
+        GroupBy.nth : Take the nth row from each group if n is an int, or a
+            subset of rows, if n is a list of ints.
+
+        Examples
+        --------
+            >>> df = pd.DataFrame([["a", 1], ["a", 2], ["a", 3], ["b", 4], ["b", 5]],
```

Comment:
these should be lef-aligned with Examples, though i don't think these examples are actually useful as they refer to the user api and NOT this function (so can just kill them here)

Reply:
I have left-aligned, but I couldn't delete them. validate_docstrings.py insists on having a non-empty examples section.

Reply:
hmm, you can simply remove the Examples section